src/main/java/training/endava/playground/generics/types/LimitedContainer.java

```diff
@@ -0,0 +1,27 @@
+package training.endava.playground.generics.types;
+
+import java.util.ArrayList;
+
+public class LimitedContainer<T, N extends Number> {
+
+    private ArrayList<T> contents;
```

Comment:
as a best practice, you should let the type of the field to be as abstract as you can, basically, you call methods just from the List interface on this field 
so, make this field reference type List